### PR TITLE
Buoyancy stabilization

### PR DESCRIPTION
**Pull request by wjhorne**
Removed buoyancyWeight inputs to Algorithm and kernel constructors and fixed issues related to accidental 6-DOF change. 

Diffs in vofBuoy are mostly due to fixes to FixedHeight user function with some differences due to stabilization. 
Diffs in generator case are due to added stabilization.

### CODE REVIEW

File: src/VolumeOfFluidEquationSystem.C [hunk 1]
```diff
@@ -117,7 +117,8 @@ VolumeOfFluidEquationSystem::VolumeOfFluidEquationSystem(
     smooth_(smooth),
     smoothIter_(smoothIter),
     isInit_(true),
-    scsAdvection_(false)
+    scsAdvection_(false),
+    buoyancyStab_(realm_.solutionOptions_->buoyancyPressureStab_ ? 1.0 : 0.0)
```
Comment (spdomin) [inline src/VolumeOfFluidEquationSystem.C:121]: I don't think you ever use this... As such, we should remove it solely in favor of using solutionOptions member data. Let me know if I am wrong.

Comment (wjhorne) [reply src/VolumeOfFluidEquationSystem.C:121]: I believe this is leftover from when I was feeding in the value into the individual kernels/algorithms. It should be removed without issue. I'll do so here in a moment and make sure it all still works

File: reg_tests/test_files/vofWaveGenerator/vofWaveGenerator.i [hunk 2]
```diff
@@ -237,11 +237,12 @@ realms:
       use_consolidated_face_elem_bc_algorithm: yes
 
       activate_balanced_force_algorithm: yes
+      activate_buoyancy_pressure_stabilization: yes
```
Comment (spdomin) [inline reg_tests/test_files/vofWaveGenerator/vofWaveGenerator.i:240]: This was to be perfectly explicit, as when BF is active, buoyancy pressure stabilization is also (similar to vofBuoy)... Keep this as I like the inclusion of this line command.

**Review by spdomin — CHANGES_REQUESTED** [review]
I think there is one last iteration - specifically removal of the buoyancy stabilization member data in VofEQS. This is never required based on the reliance to solution options.

In reality, I think you could have simply created a buoyancy weight that you pass around... Either way is fine with me.

File: src/VolumeOfFluidEquationSystem.C (hunk 1 above)
Comment (wjhorne) [reply src/VolumeOfFluidEquationSystem.C]: This has now been removed in latest push.

**Review by spdomin — APPROVED** [review]

**Pull request merged**